sklearn/ensemble/_stacking.py

```diff
@@ -0,0 +1,654 @@
+        clf1 = LogisticRegression()
+        clf2 = RandomForestClassifier()
+        eclf = StackingClassifier(estimators=[('lr', clf1), ('rf', clf2)]
+        eclf.set_params(rf=None)
+
+        """
+        super()._set_params('estimators', **params)
+        return self
+
+    def get_params(self, deep=True):
+        """ Get the parameters of the stacking estimator.
+
+        Parameters
+        ----------
+        deep: bool
+            Setting it to True gets the various classifiers and the parameters
+            of the classifiers as well.
+        """
+        return super()._get_params('estimators', deep=deep)
+
+    def _concatenate_predictions(self, predictions):
+        """Concatenate the predictions of each first layer learner.
+
+        This helper is in charge of ensuring the preditions are 2D arrays and
+        it will drop one of the probability column when the problem is binary
+        since a p(c=1) = 1 - p(c=2).
+        """
+        X_meta = []
+        for est_idx, preds in enumerate(predictions):
+            # case where the the estimator returned a 1D array
+            if preds.ndim == 1:
+                X_meta.append(preds.reshape(-1, 1))
+            else:
+                # remove one column in case of probabilities with binary case
+                if (self.predict_method_[est_idx] == 'predict_proba' and
+                        preds.shape[1] == 2):
+                    X_meta.append(preds[:, [1]])
+                else:
+                    X_meta.append(preds)
+        return np.concatenate(X_meta, axis=1)
+
+    def fit(self, X, y, sample_weight=None):
+        """Fit the estimators.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape (n_samples, n_features)
+            Training vectors, where n_samples is the number of samples and
+            n_features is the number of features.
+
+        y : array-like, shape (n_samples,)
+            Target values.
+
+        sample_weight : array-like, shape (n_samples,) or None
+            Sample weights. If None, then samples are equally weighted.
+            Note that this is supported only if all underlying estimators
+            support sample weights.
+
+        Returns
+        -------
+        self : object
+        """
+
+        self._validate_meta_estimator()
+
+        if self.estimators is None or len(self.estimators) == 0:
+            raise AttributeError(
+                "Invalid 'estimators' attribute, 'estimators' should be a list"
+                " of (string, estimator) tuples."
+            )
+
+        names, estimators_ = zip(*self.estimators)
+        self._validate_names(names)
+
+        has_estimator = False
+        for est in estimators_:
+            if est not in (None, 'drop'):
+                has_estimator = True
+
+        if not has_estimator:
+            raise ValueError(
+                "All estimators are None or 'drop'. At least one is required "
```

Comment:
I'm thinking we should only support 'drop'?

Reply:
But then we have estimator with `drop` and some other with {'drop' and None}. Would it be strange/inconsistent?

Reply:
Let's remove it then. I think Pipeline is also supporting both.